### ISSUE TITLE
Documentation is outdated, and nothing compiles

### ISSUE THREAD

**Issue by kornerr** [tower120/reactive#2]
1.
```
#include <reactive/ObservableProperty>
```
is no longer valid. Now it's
```
#include <reactive/ObservableProperty.h>
```
2. Just one line
```
#include <reactive/ObservableProperty.h>
```
already breaks the build process.

This is yet another so-called 'reactive' library I found, and none simply 'works'. What a shame.

**Comment by tower120** [comment]
Sorry for this inconvenient. What compiler do you use, and what is compiler output?

P.S. I just tried latest mingw-w64 gcc 7.1, compilation is now broken. It seems this is due to `apply` disambiguation (gcc sees it as if it be in global namespace). VS 2017 and clang 5 compiles fine.

**Comment by tower120** [comment]
Please try now, if my assumption was correct, it should fix your problems.

**Comment by kornerr** [comment]
I actually rolled out my own implementation for reporting back then in September, so this issue can be closed :)

**Comment by tower120** [comment]
Is it in public domain? Can it be watched?

**Comment by kornerr** [comment]
Yeah. It's part of 'mjin-report', part of the game tech I'm working on. Here's a sample test of its usage: https://bitbucket.org/ogstudio/mjin-report/src/tip/include/mjin-report/Test.h?at=default&fileviewer=file-view-default

Though, I ended up with reporting only without passing values, because passing values breaks when we deal with arrays: many questions arise, many corner cases, so I just limited the application to reporting only.
